### PR TITLE
Separate Embedded Zookeeper server from ha

### PR DESCRIPTION
**Pull request by yaooqinn**
yaooqinn [Closes #471](https://github.com/yaooqinn/kyuubi/pull/471)    Target Issue [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_

1. Separated embedded zk from ha module, as it's only used for setup a zk from Kyuubi server for now if there is not an external one
2. Support more configuration for the embedded zookeeper, for example, HOSTNAME, then it's more close to prod usage.
3. TODO: Make the embedded zookeeper of our own, which should support as quorum peers

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @turboFei @pan3793 @Honglun @ulysses-you

**Review by pan3793 — APPROVED** [review]

File: kyuubi-zookeeper/src/main/scala/org/apache/kyuubi/zookeeper/EmbeddedZookeeper.scala [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  override def start(): Unit = synchronized {
+    server.start()
+    info(s"$getName is started at $getConnectString")
+    sys.addShutdownHook(server.close())
+    super.start()
+  }
+
+  override def stop(): Unit = synchronized {
+    if (getServiceState == ServiceState.STARTED) {
```
Comment (yaooqinn) [inline kyuubi-zookeeper/src/main/scala/org/apache/kyuubi/zookeeper/EmbeddedZookeeper.scala:79]: with L74, can we bypass the race issue in the test with zk?  @ulysses-you

Comment (ulysses-you) [reply kyuubi-zookeeper/src/main/scala/org/apache/kyuubi/zookeeper/EmbeddedZookeeper.scala:79]: We have already added ShutdownHook in KyuubiServer ? The state check can reduce the possibility of race .. but I think it's still here.

Comment (yaooqinn) [reply kyuubi-zookeeper/src/main/scala/org/apache/kyuubi/zookeeper/EmbeddedZookeeper.scala:79]: I shall remove the hook there, but we will implement a zk of our own

**Review by turboFei — APPROVED** [review]

**Review by Honglun — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.2.0

**Pull request closed without merging**